lib/ansible/plugins/filter/bool.yml

```diff
@@ -18,6 +18,11 @@ DOCUMENTATION:
         python: Pass through the Python C(bool) function
         truthy: Just let Python evaluate in an C(if) statement
       version_added: '2.18'
+    strict:
+      description: Give an error for invalid booleans.
+      default: true
```

Comment:
@bcoca Since I restored the previous behavior, I limited this new option to the strategy `yaml`. Do you have an option on 1) should this even be a toggle 2) if so should it also apply to the new strategy types you added (even if the default is lenient)?